src/qibolab/platforms/abstract.py

```diff
@@ -134,24 +137,47 @@ def reload_settings(self):
         self.relaxation_time = settings["settings"]["repetition_duration"]
         self.sampling_rate = settings["settings"]["sampling_rate"]
 
-        # TODO: Create better data structures for native gates
+        # Load native gates
         self.native_gates = settings["native_gates"]
-        self.native_single_qubit_gates = self.native_gates["single_qubit"]
-        if "two_qubit" in self.native_gates:
-            self.native_two_qubit_gates = self.native_gates["two_qubit"]
-            for gates in self.native_gates["two_qubit"].values():
-                self.two_qubit_natives |= set(gates.keys())
-        else:
-            # dummy value to avoid transpiler failure for single qubit devices
-            self.two_qubit_natives = {"CZ"}
-
         # Load characterization settings and create ``Qubit`` and ``Channel`` objects
         for q in settings["qubits"]:
             if q in self.qubits:
                 for name, value in settings["characterization"]["single_qubit"][q].items():
                     setattr(self.qubits[q], name, value)
             else:
-                self.qubits[q] = Qubit(q, **settings["characterization"]["single_qubit"][q])
+                self.qubits[q] = qubit = Qubit(q, **settings["characterization"]["single_qubit"][q])
+                # register channels to qubits when we are using the old format
+                # needed for ``NativeGates`` to work
+                if "qubit_channel_map" in self.settings:
+                    ro, qd, qf, _ = self.settings["qubit_channel_map"][q]
+                    if ro is not None:
+                        qubit.readout = Channel(ro)
+                    if qd is not None:
+                        qubit.drive = Channel(qd)
+                    if qf is not None:
+                        qubit.flux = Channel(qf)
```

Comment:
To avoid all these checks: could we simply make such that:
```py
qubit.readout = Channel(None)
```
would result in the same `qubit` as not assigning `qubit.readout` at all?

In this way we would avoid another special case.

Reply:
I have been thinking how to do this and I only found two solutions

1. Override `Channel.__new__` to return `None` for `Channel(None)`. This is generally not good practice and may lead to other complications.
2. Accept `None` as the `Channel.name` and define a setter for `qubit.readout`. This requires changing all `Qubit` channels (`readout`, `drive`, `flux`, `feedback`, `twpa`) from simple dataclass fields to properties and implement their setters which would result to a very long `Qubit` class with some code repetition.

The current solution with the checks, even not perfect, seems to me better than the above.

Reply:
> 1. Override `Channel.__new__` to return `None` for `Channel(None)`. This is generally not good practice and may lead to other complications.

Overriding `__new__` is in general not recommended, but an available option

> Typical implementations create a new instance of the class by invoking the superclass’s [__new__()](https://docs.python.org/3/reference/datamodel.html#object.__new__) method using super().__new__(cls[, ...]) with appropriate arguments and then modifying the newly created instance as necessary before returning it.
https://docs.python.org/3/reference/datamodel.html#object.__new__

Moreover, such a use case is also accounted for

> If [__new__()](https://docs.python.org/3/reference/datamodel.html#object.__new__) does not return an instance of cls, then the new instance’s [__init__()](https://docs.python.org/3/reference/datamodel.html#object.__init__) method will not be invoked.

and if you want to modify the type of the returned object is the only available option

> Because [__new__()](https://docs.python.org/3/reference/datamodel.html#object.__new__) and [__init__()](https://docs.python.org/3/reference/datamodel.html#object.__init__) work together in constructing objects ([__new__()](https://docs.python.org/3/reference/datamodel.html#object.__new__) to create it, and [__init__()](https://docs.python.org/3/reference/datamodel.html#object.__init__) to customize it), no non-None value may be returned by [__init__()](https://docs.python.org/3/reference/datamodel.html#object.__init__); doing so will cause a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) to be raised at runtime.

However, if there are other viable solutions, I would prefer them. But if we are using `None` as a missing channel somewhere else, and there is no other reasonable alternative than checking, I would consider the `Channel(None) -> None` pattern, because it is just factorizing some repeated code reasonably (it is reasonable that if you only pass `None` you're not expecting a sensible channel out, and an empty one you can always generate with `Channel()`, if possible at all).

Reply:
Moreover, `__new__` should be untouched by `@dataclass` decorator.

Reply:
I tried implementing the `__new__` but it seems to break the pickle of this object for which we have a test. I had similar problems in the past with pickle of objects on which I was overriding the `__new__`, that's why I wanted to avoid going to that direction. Pickle can be fixed by implementing some other magic method but I am not sure if it's worth doing that.

Note that these lines exist because the `MultiqubitPlatform` does not use the `Channel` objects properly. If we migrate all drivers to the `DesignPlatform` we can probably remove this part. Checking for `None` may still appear in some places though.